src/docs/sdk/research/performance/index.mdx

```diff
@@ -192,6 +192,160 @@ In the example above, if an error bubbles up the call stack we want to be able t
+
+### Transaction Spans Gain Event Attributes
+
+When transactions are transformed from their in-memory representation to an `Event`, they gain more attributes that are not assignable to regular spans, such as `breadcrumbs`, `extra`, `contexts`, `event_id`, `fingerprint`, `release`, `environment`, `user`, etc.
+
+### Lifecycle Hooks
+
+Sentry SDKs expose a `BeforeSend` hook for error events, which allows users to modify and/or drop events before they are sent to Sentry.
+
+When the new type of event was introduced, `transaction`, it was soon decided that they would not go through the `BeforeSend` hook, for essentially two reasons:
+
+- Prevent user code from relying on the dual form of transactions (sometimes looking like a span, sometimes like an event, see earlier sections above);
+- Avoid existing `BeforeSend` functions that were written with only errors in mind from interfering with transactions, be it mutating them accidentally, dropping them altogether, or some other unexpected side-effect.
+
+However, it was also clear that some form of lifecycle hook was necessary to allow users to do things like updating a transaction name.
+
+We ended up with the middle ground of allowing the mutation/dropping of transaction events through `EventProcessor` (a more general form of `BeforeSend`), which solves problems by giving users immediate access to their data before it goes out of the SDK, but is also the worse of both worlds in that it exposes the transaction duality that was never intended to leak and is more complicated to use than `BeforeSend`.
+
+In OpenTelemetry, spans go through span processors, which are two lifecycle hooks: one when a span is started and one when it is ended.
+
+### Nested Transactions
+
+On one hand, Sentry's ingestion model does not allow for nested transactions. Transactions are meant to mark service transitions.
+
+On the other hand, Sentry as a product requires that transactions exist. Data outside of a transaction is lost.
+
+In practice, SDKs have no way of preventing transactions from becoming nested. The end result is possibly surprising to users, as each transaction starts a new tree. The only way to relate those trees is through the `trace_id`.
+
+Sentry's billing model is per event/transaction. That means that a transaction within a transaction generates two billable events.
+
+In SDKs, a transaction within a transaction causes inner spans to be "swallowed" by the innermost transaction surrounding them. In some situations, automatic instrumentation creating spans will only make them appear in one of the two transactions, but not both, causing apparent gaps in instrumentation.
```

Comment:
This is only the case because 1) the SDKs split spans by transaction (and they must do it) and 2) the UI (at the moment) only shows a single transaction at a time, ignoring anything that may exist around it in the same trace.